docs/03_features/pydantic.md

```diff
@@ -0,0 +1,153 @@
+---
+title: Pydantic Support
+path: /docs/feature/pydantic
+---
+
+# Pydantic support
+
+Strawberry comes with support for
+[Pydantic](https://pydantic-docs.helpmanual.io/). This allows to create
+Strawberry types from Pydantic models without having to write code twice.
+
+Here's a basic example of how this works, let's say we have a Pydantic Model for
+a user, like this:
+
+```python
+from datetime import datetime
+from typing import List, Optional
+from pydantic import BaseModel
+
+
+class User(BaseModel):
+    id: int
+    name: str
+    signup_ts: Optional[datetime] = None
+    friends: List[int] = []
+```
+
+We can create a Strawberry type by using the `strawberry.beta.pydantic.type`
+decorator:
+
+```python
+import strawberry
+
+from .models import User
+
+@strawberry.beta.pydantic.type(model=User, fields=[
+    'id',
+    'name',
+    'friends'
+])
+class UserType:
+    pass
+```
+
+The `strawberry.beta.pydantic.type` decorator accepts a Pydantic model and a list of
+fields that we want to expose on our GraphQL API.
+
+> **Note** specifying the list of field is required to prevent accidentally
+> exposing fields that weren't meant to be exposed on a API
+
+## Input types
+
+Input types are similar to types, we can create one by using the
+`strawberry.beta.pydantic.input` decorator:
+
+```python
+import strawberry
+
+from .models import User
+
+@strawberry.beta.pydantic.input(model=User, fields=[
+    'id',
+    'name',
+    'friends'
+])
+class UserType:
```

Comment:
Another point here, if the name of the class is what it ends up being in the schema, I don't think the docs should include examples with names like `FooType` and `FooInputType`, should just be `Foo` and `FooInput`. If it clashes with an existing namespace name like in this case, we should alias the original.

Reply:
Not sure I agree with that @AndrewIngram . I think adding `Type` to the end of the type name is fine. It's not ideal but it's better than having to deal with any confusion around which thing you're dealing with.

Reply:
People have a tendency to copy and adapt examples from docs because they see them as a fountain of "correct" examples (the React team is trying to tackle this problem right now), so I fear that this is going to result in people new to GraphQL end up with `FooTypes` littered throughout their schema and not realise that it's not really idiomatic GraphQL.

Reply:
👌